Kernel/FileSystem/FATFS/Inode.cpp

```diff
@@ -576,8 +576,15 @@ ErrorOr<size_t> FATInode::write_bytes_locked(off_t offset, size_t size, UserOrKe
     dbgln_if(FAT_DEBUG, "FATInode[{}]::write_bytes_locked(): Writing size: {} offset: {}", identifier(), size, offset);
 
     u32 new_size = max(m_entry.file_size, offset + size);
-    if (new_size != m_entry.file_size)
-        TRY(resize(new_size));
+    if (new_size != m_entry.file_size) {
+        Optional<u64> clear_from;
+        Optional<u64> to_clear;
+        if (offset > m_entry.file_size) {
```

Comment:
Shouldn't this check involve `size` as well somehow? Can't `offset` be in the old file range, but `offset + size` be outside?

Reply:
Oh right, that makes sense